OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -133,21 +130,52 @@ public OpenAlSoundEngine(string deviceName)
 			context = ALC10.alcCreateContext(device, null);
 			if (context == IntPtr.Zero)
 				throw new InvalidOperationException("Can't create OpenAL context");
+
 			ALC10.alcMakeContextCurrent(context);
 
-			for (var i = 0; i < PoolSize; i++)
+			var size = new int[1];
+			ALC10.alcGetIntegerv(device, ALC10.ALC_ATTRIBUTES_SIZE, 1, size);
+			if (!CheckError($"Failed to query OpenAL attributes using {deviceName}"))
 			{
-				AL10.alGenSources(1, out var source);
-				if (AL10.alGetError() != AL10.AL_NO_ERROR)
+				var values = new int[size[0]];
+				ALC10.alcGetIntegerv(device, ALC10.ALC_ALL_ATTRIBUTES, size[0], values);
+				if (!CheckError($"Failed to query OpenAL attributes using {deviceName}"))
 				{
-					Log.Write("sound", $"Failed generating OpenAL source {i}");
-					return;
+					for (var i = 0; i < values.Length; i++)
+						if (values[i] == ALC11.ALC_MONO_SOURCES || values[i] == ALC11.ALC_STEREO_SOURCES)
+							if (values[i + 1] > poolSize)
```

Comment:
Shouldn't `poolSize` default to `0` then? There might be devices which report less than 32.

Reply:
Did we have such reports in the last 14 years?

Reply:
Nobody explicitly said this is the issue, but who would know such a detail when the audio isn't working as expected. Why do we want to risk doing something wrong? Please just change the default, it will not matter in practice but then we are on the save side.

Reply:
ok

Reply:
We probably also want to clamp this to a sensible maximum - we're never going to need 65535 sources, and there is likely to be a performance and/or memory overhead for larger counts.